src/AmalgamatorSession.ts

```diff
@@ -675,6 +707,20 @@ export class AmalgamatorSession extends LoggingDebugSession {
             ].customRequest('cdt-gdb-adapter/Memory', args);
             response.body = childResponse.body;
             this.sendResponse(response);
+        } else if (command === 'cdt-amalgamator/resumeAll') {
+            const [, threads] = await this.collectChildTheads();
+            threads.forEach(async (thread) => {
+                const [childIndex, childId] = await this.getThreadInfo(
+                    thread.id
+                );
+                args = { threadId: childId } as DebugProtocol.ContinueArguments;
+                await this.childDaps[childIndex].customRequest(
+                    'cdt-gdb-adapter/resumeAll',
```

Comment:
The change in https://github.com/eclipse-cdt-cloud/cdt-gdb-adapter/pull/255 sends a continue event to all thread regardless of whether they are running, this leads to error like this in the log. 

`
To client: {"seq":126,"type":"event","event":"output","body":{"category":"stdout","output":"GDB result: 18 error,msg=\"Cannot execute this command while the selected thread is running.\"\n"}}
`

Note I only see the above when verbose is on because the error is actually lost.

Reply:
I have removed the event submission here. However, in this repository, we need to check the thread states at the `continueRequest()` function before sending `mi.sendExecContinue()` to GDB. The purpose of this is to avoid the error `'Cannot execute this command while the selected thread is running.'`

Reply:
I removed the customRequest() event and called `continueRequest() ` instead at cdt-amalgamator. And check the thread states at the` continueRequest() ` function before sending `mi.sendExecContinue()`  to GDB in cdt-gdb-adapter. The purpose of this is to avoid the error 'Cannot execute this command while the selected thread is running.'

Reply:
I think that check should happen in the amalgamator, not in the adapter. If you have good reason for it to be in the adapter instead, please expand as to why you can't do it in the amalgamator.

Reply:
Hi @jonahgraham ,
I updated the event status check in cdt-amalgamator.
Could you please help me review it?

Thank you!